### PR TITLE
Cext 2742 - CLI status command changes related to edge.

### PR DESCRIPTION
**Pull request by nvyasadobe**
CLI status command changes related to edge.

## Description

This PR contains changes related to mesh status cli command. Including the status of edge mesh as well.

## Related Issue

https://jira.corp.adobe.com/browse/CEXT-2742

## Motivation and Context

This is required to show edge mesh and legacy both mesh status.

## How Has This Been Tested?

By linking CLI in local setup and verifying against the mesh locally created.

## Screenshots (if appropriate):
(if  showEdgeMeshUrl returns true)
image
(if  showEdgeMeshUrl returns false)
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/commands/api-mesh/status.js [hunk 1]
```diff
@@ -35,7 +42,47 @@ class StatusCommand extends Command {
...
 				const mesh = await getMesh(imsOrgId, projectId, workspaceId, workspaceName, meshId);
+				if (showEdgeMeshUrl) {
+					const meshDeployments = await getMeshDeployments(
+						imsOrgCode,
+						projectId,
+						workspaceId,
+						meshId,
+					);
+					this.log(chalk.blackBright('Edge Mesh Status:'));
+					switch (String(meshDeployments.status).toLowerCase()) {
```
Comment (nvyasadobe) [inline src/commands/api-mesh/status.js:55]: Added `String(meshDeployments.status).toLowerCase()` to handle `SUCCESS`, `success` and `Success`. If the entry in table would always be unique(lowercase) we can remove this.

File: src/lib/devConsole.js [hunk 2]
```diff
@@ -886,6 +886,64 @@ const getTenantFeatures = async organizationCode => {
...
+ * @param projectId
+ * @param workspaceId
+ * @param meshId
+ * @returns {Promise<Object>}
+ */
+const getMeshDeployments = async (organizationCode, projectId, workspaceId, meshId) => {
+	const { accessToken, apiKey } = await getDevConsoleConfig();
+	const config = {
+		method: 'get',
+		url: `${SMS_BASE_URL}/organizations/${organizationCode}/projects/${projectId}/workspaces/${workspaceId}/meshes/${meshId}/deployments/latest?API_KEY=${apiKey}`,
```
Comment (dthampy) [inline src/lib/devConsole.js:905]: Quick question @nvyasadobe, So here do we eventually plan to route through DEV_CONSOLE_BASE_URL and you have SMS_BASE_URL for testing for the time being?

**Review by dthampy — APPROVED** [review]

File: src/lib/devConsole.js (hunk 2 above)
Comment (brasewel) [reply src/lib/devConsole.js:905]: No, these will always go directly to SMS.
The new tenants route and deployments route bypass devconsole. We are moving away from devconsole

File: src/commands/api-mesh/status.js [hunk 3]
```diff
@@ -35,7 +42,58 @@ class StatusCommand extends Command {
...
+								'******************************************************************************************************',
+							);
+							break;
+						default:
+							//Case when deployments table has no status and only initial metadata
+							this.log(
+								'******************************************************************************************************',
+							);
+							this.log(
+								'Your mesh is not having any status. Please wait for a while and try again.',
```
Comment (brasewel) [inline src/commands/api-mesh/status.js:89]: This messaging is grammatically incorrect. I would message Naomi to get proper messaging on this

**Review by brasewel — CHANGES_REQUESTED** [review]

File: src/commands/api-mesh/status.js [hunk 4]
```diff
@@ -35,7 +42,58 @@ class StatusCommand extends Command {
...
+						workspaceId,
+						meshId,
+					);
+					this.log(chalk.blackBright('Edge Mesh Status:'));
+					switch (String(meshDeployments.status).toLowerCase()) {
+						case 'success':
+							this.log(
+								'******************************************************************************************************',
+							);
+							this.log('Your mesh has been successfully built.');
```
Comment (jhadobe) [inline src/commands/api-mesh/status.js:60]:
```suggestion
							this.log('Your mesh was successfully built.');
```

File: src/commands/api-mesh/status.js [hunk 5]
```diff
@@ -35,7 +42,58 @@ class StatusCommand extends Command {
...
+							this.log('Your mesh is provisioning.');
+							this.log(
+								'******************************************************************************************************',
+							);
+							break;
+						case 'error':
+							this.log(
+								'******************************************************************************************************',
+							);
+							this.log('Your mesh errored out with the following error. ', meshDeployments.error);
```
Comment (jhadobe) [inline src/commands/api-mesh/status.js:78]:
```suggestion
							this.log('Your mesh encountered the following error: ', meshDeployments.error);
```

File: src/commands/api-mesh/status.js (hunk 3 above)
Comment (jhadobe) [inline src/commands/api-mesh/status.js:89]:
```suggestion
								'Your mesh status is not available. Please wait for a while and try again.',
```

**Review by jhadobe — CHANGES_REQUESTED** [review]

File: src/commands/api-mesh/status.js (hunk 3 above)
Comment (nvyasadobe) [reply src/commands/api-mesh/status.js]: Have updated the messages as per the suggestion of Jared.

**Review by brasewel — APPROVED** [review]

**Review by amolina-adobe — APPROVED** [review]

**Pull request merged**